lib/pairing-service.js

```diff
@@ -51,15 +52,15 @@ pairingService.pollPairingStatus = () => {
                 logger.info('Checking pairing status, attempt ' + (++counter) +
                     ' of ' + MAXIMUM_TRIES);
                 pairingService.getPairingStatus()
-                    .then(() => {
+                    .then((response) => {
```

Comment:
We don't need to propagate this information through all the layers. This way we are creating unnecessary spaghetti

Reply:
It's better to return what the core returns and leave the final implementation to end consumer of the method to interpret the situation based on requirement.